pynhm/base/storageUnit.py

```diff
@@ -88,39 +67,37 @@ def finalize(self) -> None:
         self._finalize_netcdf()
 
     @staticmethod
-    def get_required_parameters() -> list:
+    def get_parameters() -> list:
         raise Exception("This must be overridden")
 
-    def initialize_output_data(self):
-        self.output_column_names = ["date"]
-        if "nhm_id" in self.params.parameters:
-            self.output_column_names += [
-                f"nhru_hru_intcpstor_{nhmid}"
-                for nhmid in self.params.parameters["nhm_id"]
-            ]
-        else:
-            self.output_column_names += [
-                f"intcpstor_{id}" for id in range(self.nhru)
-            ]
-
-        # Initialize the output_data dictionary for each entry with an empty list
-        self.output_data = {}
-        for output_name in self.output_data_names:
-            self.output_data[output_name] = []
-        return
+    @staticmethod
+    def get_input_variables() -> list:
+        raise Exception("This must be overridden")
 
-    def get_output_dataframes(self):
-        """
-        Return a dictionary of data frames of output data from this process
+    @staticmethod
+    def get_variables() -> list:
+        raise Exception("This must be overridden")
 
-        """
-        output_data = {}
-        for key in self.output_data:
-            value = self.output_data[key]
-            df = pd.DataFrame(value, columns=self.output_column_names)
-            df.set_index("date", inplace=True)
-            output_data[key] = df
-        return output_data
+    @property
+    def parameters(self) -> list:
+        return self.get_parameters()
+
+    @property
+    def input_variables(self) -> list:
+        return self.get_input_variables()
+
+    @property
+    def variables(self) -> list:
+        return self.get_variables()
```

Comment:
I'm not sure I love different names for static methods and another name for properties. 
I DO love the brevity of the property names.